### DISCUSSION TITLE
Hide parameter doesnt work for Containable

### DISCUSSION THREAD

**Discussion by PanmanS** [FakeFishGames/Barotrauma#15192] [Ideas and suggestions]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [X] My issue happened while using mods.

### What happened?

I have section of code below that I use for my laser gun and it has this line:  `<Containable items="lasergun_round_co" hide="true" />`.  It doesnt work at all.
```

    <ItemContainer capacity="1" maxstacksize="1" slotsperrow="2" hideitems="false" iteminterval="0,-4" itempos="27,10" containedstateindicatorstyle="battery" containedstateindicatorslot="1" spawnwithid="lasergun_round_co" containedspritedepth="0.56">
      <Containable items="lasergun_round_co" hide="true" />
      <SubContainer capacity="1" maxstacksize="1">
        <Containable items="laser_magazine_co" hide="false" itempos="-1,-5" rotation="23" ExcludeBroken="false">
          <StatusEffect type="OnInserted" forceplaysounds="true" target="Contained">
            <Sound file="%ModDir%/Sounds/light_in.ogg" range="300" loop="false" />
          </StatusEffect>
          <StatusEffect type="OnRemoved" forceplaysounds="true" target="Contained">
            <Sound file="%ModDir%/Sounds/light_out.ogg" range="300" loop="false" />
          </StatusEffect>
        </Containable>
      </SubContainer>
      <SubContainer capacity="1" maxstacksize="1">
        <Containable items="attachment_light" setactive="true" itempos="11,2" ExcludeBroken="false">
          <StatusEffect type="OnInserted" forceplaysounds="true" target="Contained">
            <Sound file="%ModDir%/Sounds/light_in.ogg" range="300" loop="false" />
          </StatusEffect>
          <StatusEffect type="OnRemoved" forceplaysounds="true" target="Contained">
            <Sound file="%ModDir%/Sounds/light_out.ogg" range="300" loop="false" />
          </StatusEffect>
        </Containable>
      </SubContainer>
      <SlotIcon slotindex="1" texture="%ModDir%/magazineicons.png" sourcerect="872,55,240,240" origin="0.45,0.5" />
      <SlotIcon slotindex="2" texture="Content/UI/CampaignUIAtlas2.png" sourcerect="896,536,128,64" origin="0.45,0.5" />
    </ItemContainer>

```
Here's how it looks in-game (3rd slot not supposed to be shown):
![Barotrauma_ZH0NinUF9J](https://github.com/user-attachments/assets/ba34ef08-78ba-4470-ae28-6ce3ad4315cf)

You can only hide a slot if it is a separate ItemContainer, but then you can't drag & drop things into it and StatusEffects can't check for stuff in it and that fucking sucks.

### Reproduction steps

1. create any item with ItemContainer and do spawnwithid="batterycell",
2. add `<Containable items="batterycell" hide="true" />`
3. test it in-game

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Happens in both single player and multiplayer

### Version

v1.6.19.1 (Unto the Breach Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by Regalis11** [comment]
To my knowledge there's no "hide" parameter, nor a way to only hide specific slots of an inventory, so this seems more like a feature request than a bug report to me. But I agree, this would be a useful feature.

**Reply by PanmanS** [reply]
huh, wait but............ Ig Im :baromodder:

**Reply by UnhappyWithMyUsername** [reply]
Yes please!

**Comment by ViktorShield** [comment]
I second this. Had a hard time with exosuits because you can't hide specific subcontainers. Say, if you want some item to only have 1 slot visible and 1 slot hidden. For example, in vanilla Funbringer (clown exosuit), there are 2 ItemContainer. While it technically works for this one, you cannot do anything slot-specific to these inventories since their slots are the same. 

Example:
```
        <ItemContainer/>
...
        <ItemContainer/>
...
        <StatusEffect type="OnUse" target="Contained" TargetSlot="0">
          <Remove/>
```

This will target slot "1" in BOTH parts of 1 inventory

So we modders (and devs AFAIS) have to either show our dirty xml laundry to public by not hiding the items in our very secret subcontainers, or to deal with multiple ItemContainers and both options suck for obvious reasons
